src/Form/PasswordChange.php

```diff
@@ -0,0 +1,163 @@
+    $this->dateTime = $date_time;
+    $this->user = $user;
+    $this->password = $password;
+    $this->sessionManager = $session_manager;
+  }
+
+  /**
+   * {@inheritdoc}
+   */
+  public static function create(ContainerInterface $container): PasswordChange {
+    $user_id = $container->get('current_user')->id();
+    return new static(
+      $container->get('datetime.time'),
+      $container->get('entity_type.manager')->getStorage('user')->load($user_id),
+      $container->get('password'),
+      $container->get('session_manager')
+    );
+  }
+
+  /**
+   * {@inheritdoc}
+   */
+  public function getFormId(): string {
+    return 'password_enhancements_change_form';
+  }
+
+  /**
+   * {@inheritdoc}
+   */
+  public function buildForm(array $form, FormStateInterface $form_state): array {
+    // Get user roles.
+    $roles = $this->user->getRoles();
+    $form_state->setValue('roles', $roles);
+    $form_state->setUserInput(['roles' => $roles]);
+
+    $user_pass_reset = FALSE;
+    if (!$form_state->get('user_pass_reset') && ($token = $this->getRequest()->get('pass-reset-token'))) {
+      $session_key = 'pass_reset_' . $this->user->id();
+      $user_pass_reset = isset($_SESSION[$session_key]) && Crypt::hashEquals($_SESSION[$session_key], $token);
+      $form_state->set('user_pass_reset', $user_pass_reset);
+    }
+
+    $form['account'] = [
+      '#type' => 'container',
+      'roles' => [
+        '#type' => 'hidden',
+        '#value' => $roles,
+      ],
+      'current_pass' => [
+        '#type' => 'password',
+        '#title' => $this->t('Current password'),
+        '#required' => TRUE,
+        '#attributes' => ['autocomplete' => 'off'],
+        '#access' => !$user_pass_reset,
+      ],
+      'pass' => [
+        '#type' => 'password_confirm',
+        '#size' => 25,
+        '#required' => TRUE,
+      ],
+    ];
+
+    $form['actions'] = [
+      '#type' => 'container',
+      'save' => [
+        '#type' => 'submit',
+        '#value' => $this->t('Change'),
+      ],
+    ];
+
+    return $form;
+  }
+
+  /**
+   * {@inheritdoc}
+   */
+  public function validateForm(array &$form, FormStateInterface $form_state): void {
+    parent::validateForm($form, $form_state);
+
+    if (!$form_state->get('user_pass_reset') && !$this->password->check($form_state->getValue('current_pass'), $this->user->getPassword())) {
+      $form_state->setError($form['account']['current_pass'], $this->t('Incorrect password.'));
+    }
+  }
+
+  /**
+   * {@inheritdoc}
+   */
+  public function submitForm(array &$form, FormStateInterface $form_state): void {
+    // Remove lock and clean-up session variables.
+    $attributes_bag = $this->sessionManager->getBag('attributes')->getBag();
+    $attributes_bag->remove('password_enhancements_password_change_required');
+    $attributes_bag->remove('password_enhancements_pass_reset_token');
+
+    // Update user fields.
+    $this->user->set('password_change_required', FALSE)
```

Comment:
I guess these `password_change*` fields should be changed also when user's password changes with Drush, programmatically or via a different form. I would like to recommend to use `hook_user_*` hooks to update these field values.
Also, probably the name of these fields should be more unique, should contain the module's name or its abbreviation.

Reply:
Actually the the `password_changed` field doesn't have to be updated here as it is handled with hook_entity_presave if the password was changed.

Although I think the `password_change_required` should be changed here only, because this is a setting for the user to require password change, probably wouldn't be a good idea to change it if the password would be changed via drush.

I'll change the fields' names.